src/eth.rs

```diff
@@ -315,6 +317,8 @@ impl TxRing {
         let addr = &self.desc_buf[self.cur_desc] as *const _ as u32;
         assert_eq!(addr & 0x3, 0);
         let dma = unsafe { pac::Peripherals::steal().ETHERNET_DMA };
+        // Ensure change to desc3 OWN flag occurs before tail pointer adjustment
```

Comment:
To be picky, I think it's not just the OWN flag, but also the packet buffer contents. (It is implied from context, i.e. source code order, though.)